_articles/mobile/files.md

```diff
@@ -0,0 +1,71 @@
+
+Using the [Java REST SDK](/articles/java.html), this can be broken down into two steps. First, developers need to create a metadata object describing your file in all its revisions:
+
+```java
+ForDevelopersApi devsApi = developer.getClient(ForDevelopersApi.class);
+
+FileMetadata meta = new FileMetadata();
+meta.setName("Name");
+meta.setDescription("Description");
+
+final GuidVersionHolder keys = devsApi.createFile(meta).execute().body();
+meta.setGuid(keys.getGuid());
+meta.setVersion(keys.getVersion());
+```
+
+Once this object exists and you have the GUID of the file, the Java SDK contains a utility method to perform the creation of a single revision for that file:
+
+```java
+File file = new File("files/my-file.pdf");
+
+FilesApi filesApi = developer.getClient(FilesApi.class);
+String downloadURL = RestUtils.uploadHostedFileToS3(filesApi, meta.getGuid(), file);
+```
+
+This includes the upload of the file's contents to S3, so the downloadURL that is returned is the URL to retrieve the file via HTTP.
+
+## Participant Data API
+
+Arbitrary JSON data for a participant can be [stored under an identifying key](/swagger-ui/index.html#/Participant%20Data/saveDataForSelf) through the Participant Data APIs. [A list of file identifiers can be retrieved,](/swagger-ui/index.html#/Participant%20Data/getAllDataForSelf) and these can then be used to [retrieve JSON under that key.](/swagger-ui/index.html#/Participant%20Data/getDataByIdentifierForSelf) Participants can also [delete their own files.](/swagger-ui/index.html#/Participant%20Data/deleteDataByIdentifier) 
+
+Unlike the other APIs reviewed here, these uploads and downloads of the JSON is made in one call to the Bridge server (it is not stored or downloaded through a separate call to S3 using a presigned URL).
+
+This API can be contrasted to the [participant reports API](/swagger-ui/index.html#/Participant%20Reports) which also stores arbitrary JSON data, but was designed for storing and receiving time series data (like daily reports).
+
+The data APIs have additional APIs for workers and administrators (not discussed here). The data is not available to researchers or study coordinators. 
+
+## Participant Files API
+
+[Storing data files for a participant](/swagger-ui/index.html#/Participant%20Data) involves the following steps:
+
+1. Caller creates a [ParticipantFile](/model-browser.html#ParticipantFile) record through the [create participant file API.](/swagger-ui/index.html#/Participant%20Files/createParticipantFile) Please note that the mime/type for the file is required, although Bridge will currently allow you to create a file record without it;
+1. The returned ParticipantFile that is returned will have an `uploadURL` field with a presigned URL that can be used to `PUT` the file to S3;
+1. Client uploads the file to S3 (this call will fail if the file already exists);
```

Comment:
Two nitpicks:

1. It's not the call to S3 that fails, but the call to Bridge that fails.
2. This is considered a bug and we have a Jira to fix this https://sagebionetworks.jira.com/browse/BRIDGE-2926 Should we mention this restriction in the docs at all? And if we do, should we mention that the restriction is unintentionally and will be removed?

Reply:
Moved this and added an explanation that currently, the file cannot be overwritten, but we have plans to change this so the file can be updated.